src/main/java/com/shapesecurity/shift/parser/Parser.java

```diff
@@ -81,11 +85,27 @@ private void consumeSemicolon() throws JsError {
 
     @NotNull
     protected <T extends Node> T markLocation(@NotNull SourceLocation startLocation, @NotNull T node) {
-        // TODO: actually mark location
-//    node.loc = Maybe.just(new SourceSpan(Maybe.nothing(), startLocation, new SourceLocation(this.lastLine+1, this.lastIndex-this.lastLineStart, this.lastIndex)));
+        locations.put(
+                node,
```

Comment:
At first I did not want to modify the structure itself, since it was not my code. Introducing a new variable, `endLocation` does make it easier to read, thanks. The next commit will have this modification.